src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -876,8 +883,25 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
             });
 
             // Connect the cluster to the original node and port
-            var connector = ConnectorModel.Make(node.NodeModel, targetNodeFromCluster.NodeModel, 0, ClusterResultItem.EntryNodeInPort);
-            newNodesAndWires.Add(connector);
+            try
+            {
+                dynamoViewModel.Model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(node.NodeModel.GUID, 0,
+                    PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin));
+                dynamoViewModel.Model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(targetNodeFromCluster.NodeModel.GUID, ClusterResultItem.EntryNodeInPort,
+                    PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End));
+                //remove the connector creation from undo, we handle that below
+                wsViewModel.Model.UndoRecorder.PopFromUndoGroup();
```

Comment:
Before this can we check that a connection command was actually successful ? Otherwise we will be popping the wrong thing.